### ISSUE TITLE
displayName not automatically set

### ISSUE THREAD

**Issue by Ahrengot** [jsdf/coffee-react#11]
Hey there — Thanks for creating this compiler. I'm blown away with how it just works seamlessly and gives me the best from both CoffeeScript and React.js. One feature is missing though:

The official JSX syntax sets the [`displayName`](http://facebook.github.io/react/docs/component-specs.html#displayname) property automatically. This behavior isn't carried over to coffee-react which makes debugging with the react debug tools harder because you manually have to set this property on every single component.

See this ticket on the official react.js repo for more info: https://github.com/facebook/react/issues/2447

**Comment by mjpizz** [comment]
just ran into this myself :) @jsdf I agree it would be nice to bundle the transform into the package.

My current workaround is adding the non-coffee [reactify](https://github.com/andreypopp/reactify) transform to my pipeline. It looks something like this:

```coffee
patchDisplayNameUsingReactify = (filename, options) ->
  optionsForcingCompilationOfCJSX = _.extend({extension: ["cjsx"]}, options)
  reactify(filename, optionsForcingCompilationOfCJSX)

bundle = browserify()
bundle.transform("coffee-reactify")
bundle.transform(patchDisplayNameUsingReactify)
```

You may be able to just incorporate [ReactTools.transform](https://www.npmjs.com/package/react-tools) directly.

**Comment by noinkling** [comment]
+1

Because I'm using sprockets I can just give my files the extension `.jsx.coffee.cjsx`, but it would be nice to see it integrated for parity.

**Comment by moelee** [comment]
+1
Using the workaround @noinkling suggested as well. Would be great to have this included in coffee-react.

**Comment by jsdf** [comment]
Fixed by [4.0.0](https://github.com/jsdf/coffee-react/commit/f9e5d5864e1ca566e6eeb0bed82c4ade25442044)
